From 35c0c8054d6b06cb564516eb48586b21362dfb7d Mon Sep 17 00:00:00 2001
From: Matt Wittmann <mwittmann@relaytx.com>
Date: Mon, 6 Mar 2023 08:09:00 -0800
Subject: [PATCH 2/3] Adapt cmake build

* Use deps from nix

* Parameterize python version

* Remove -Werror (warnings from eigen)
---
 timemachine/cpp/CMakeLists.txt | 25 +++++++------------------
 1 file changed, 7 insertions(+), 18 deletions(-)

diff --git a/timemachine/cpp/CMakeLists.txt b/timemachine/cpp/CMakeLists.txt
index 6c8a5dc4..60e27fa4 100644
--- a/timemachine/cpp/CMakeLists.txt
+++ b/timemachine/cpp/CMakeLists.txt
@@ -2,14 +2,16 @@ cmake_minimum_required(VERSION 3.24 FATAL_ERROR)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(PYBIND11_PYTHON_VERSION 3.10 CACHE STRING "Which version of python we're building wrappers against")
+set(PYBIND11_PYTHON_VERSION @pythonVersion@ CACHE STRING "Which version of python we're building wrappers against")
 
 project(timemachine LANGUAGES CXX CUDA)
 
-find_package(PythonInterp 3.10 REQUIRED)
-find_package(PythonLibs 3.10 REQUIRED)
+find_package(PythonInterp @pythonVersion@ REQUIRED)
+find_package(PythonLibs @pythonVersion@ REQUIRED)
+find_package(Eigen3 3.4 REQUIRED NO_MODULE)
+find_package(pybind11)
 
-string(APPEND CMAKE_CXX_FLAGS "-Werror -Wall -Wextra")
+string(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra")
 
 string(APPEND CMAKE_CUDA_FLAGS "-O3 -lineinfo")
 string(APPEND CMAKE_CUDA_FLAGS
@@ -29,19 +31,6 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 endif()
 
 set(LIBRARY_NAME custom_ops)
-set(PYBIND_SRC_DIR pybind11)
-
-if(NOT EXISTS ${PYBIND_SRC_DIR})
-  execute_process(COMMAND git clone --branch v2.9.2 https://github.com/pybind/pybind11.git ${PYBIND_SRC_DIR})
-endif()
-
-set(EIGEN_SRC_DIR eigen)
-if(NOT EXISTS ${EIGEN_SRC_DIR})
-  execute_process(COMMAND git clone --branch 3.3.9 https://gitlab.com/libeigen/eigen.git ${EIGEN_SRC_DIR})
-endif()
-
-add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/${PYBIND_SRC_DIR} ${CMAKE_CURRENT_BINARY_DIR}/${PYBIND_SRC_DIR})
-
 
 # Disable warnings on vendored files
 file(
@@ -115,7 +104,7 @@ pybind11_add_module(${LIBRARY_NAME} SHARED NO_EXTRAS
 
 include_directories(src/kernels)
 include_directories(SYSTEM ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
-include_directories(SYSTEM ${CMAKE_CURRENT_BINARY_DIR}/${EIGEN_SRC_DIR})
+include_directories(${EIGEN3_INCLUDE_DIRS})
 include_directories(SYSTEM ${CMAKE_CURRENT_BINARY_DIR}/${CUB_SRC_DIR})
 
 set_property(TARGET ${LIBRARY_NAME} PROPERTY CUDA_STANDARD 17)
-- 
2.42.0

